src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +279,837 @@ static DamageTable D_808792D4 = {
+        func_800B9010(&this->actor, NA_SE_EN_DODO_J_BREATH - SFX_FLAG);
+    }
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        func_80877D50(this);
+    }
+}
+
+void func_80877D50(EnDodongo* this) {
+    SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, &D_06003088);
+    this->actionFunc = func_80877D90;
+}
+
+void func_80877D90(EnDodongo* this, GlobalContext* globalCtx) {
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        func_808773C4(this);
+        this->timer = Rand_S16Offset(10, 20);
+    }
+}
+
+void func_80877DE0(EnDodongo* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_060028F0, -1.0f, 35.0f, 0.0f, 2, -4.0f);
+    this->actor.flags |= 0x10;
+    this->timer = 25;
+    this->actionFunc = func_80877E60;
+    this->actor.speedXZ = 0.0f;
+}
+
+void func_80877E60(EnDodongo* this, GlobalContext* globalCtx) {
+    static Vec3f D_80879360 = { 0.0f, 0.6f, 0.0f };
+    static Color_RGBA8 D_8087936C = { 255, 255, 255, 255 };
+    Vec3f sp84;
+    Vec3f sp78;
+    s16 i;
+    Actor* explosive = globalCtx->actorCtx.actorList[ACTORCAT_EXPLOSIVES].first;
+    Vec3f sp64;
+    s32 pad;
+    s16 sp5E;
+    s16 sp5C;
+
+    while (explosive != NULL) {
+        if (explosive == this->actor.child) {
+            break;
+        }
+        explosive = explosive->next;
+    }
+
+    if (explosive == NULL) {
+        this->actor.child = NULL;
+    }
+
+    if (this->actor.child != NULL) {
+        Math_Vec3f_Copy(&this->actor.child->world.pos, &this->unk_308);
+        this->actor.child->freezeTimer = 30000;
+    }
+
+    if (func_801378B8(&this->skelAnime, 28.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_J_EAT);
+        if (this->actor.child != NULL) {
+            Actor_MarkForDeath(this->actor.child);
+            this->actor.child = NULL;
+        }
+        this->actor.flags &= ~0x10;
+    } else if (this->skelAnime.animPlaybackSpeed > -0.5f) {
+        this->timer--;
+        if (this->timer == 10) {
+            for (i = 10; i >= 0; i--) {
+                sp84.x = randPlusMinusPoint5Scaled(10.0f);
+                sp84.y = randPlusMinusPoint5Scaled(10.0f);
+                sp84.z = randPlusMinusPoint5Scaled(10.0f);
+                sp78.x = sp84.x * -0.1f;
+                sp78.y = sp84.y * -0.1f;
+                sp78.z = sp84.z * -0.1f;
+                sp64.x = this->collider1Elements[0].dim.worldSphere.center.x + sp84.x;
+                sp64.y = this->collider1Elements[0].dim.worldSphere.center.y + sp84.y;
+                sp64.z = this->collider1Elements[0].dim.worldSphere.center.z + sp84.z;
+                func_800B0EB0(globalCtx, &sp64, &sp84, &sp78, &this->unk_32C, &this->unk_330, this->unk_334 * 400.0f,
+                              this->unk_334 * 10.0f, 10);
+            }
+
+            Audio_PlayActorSound2(&this->actor, NA_SE_IT_BOMB_EXPLOSION);
+            if (this->actor.colChkInfo.health < 5) {
+                this->actor.colChkInfo.health = 0;
```

Comment:
```suggestion
            if (this->actor.colChkInfo.health <= 4) {
                this->actor.colChkInfo.health = 0;
```
Since `-= 4` is used below